plugins/modules/net_tools/nmcli.py

```diff
@@ -1078,13 +1089,18 @@ def connection_update(self, nmcli_command):
 
         # Constructing the command.
         for key, value in options.items():
+            if key in self.WIFI_SEC_SECRET_OPTIONS:
+                self.edit_commands += ['set %s %s' % (key, value)]
+                continue
             if value is not None:
                 cmd.extend([key, value])
 
         return self.execute_command(cmd)
 
     def create_connection(self):
         status = self.connection_update('create')
+        if status[0] == 0 and self.edit_commands:
+            status = self.edit_connection()
         if self.create_connection_up:
```

Comment:
Should we add something similar here too?
```suggestion
        if status[0] == 0 and self.create_connection_up:
```

Reply:
Not in this PR IMO. Maybe a future item to consider, but not necessary for this fix.